main.bal

```diff
@@ -15,36 +15,45 @@ service on new tcp:Listener(3000) {
         return new Service();
     }
 }
+string firstMessage = "";
+string secondMessage = "";
 
 service class Service {
 //once the content is received from the client, this method is invoked
 
+    byte[] buffer = [];
+    byte[] buffer1 = [];
+    int? expectedLen = ();
+
     remote function onBytes(tcp:Caller caller, readonly & byte[] data) 
-        returns tcp:Error? {
-        // io:println(string:fromBytes(data));  
+        returns tcp:Error? { 
 
-        string|error firststring = string:fromBytes(data);   
+            // io:println(data);
 
-            if firststring is string{
-                string finalstring = regex:replaceAll(firststring, "/[^ -~]+/g", "");
-                int x = <int> finalstring.indexOf("{");
-                int y = finalstring.length();
-                string secondstring = finalstring.substring(x,y);
-                
-                // io:println(secondstring);
+            byte[] byteArray = data;
+            int? lastIndex = byteArray.lastIndexOf(10);
+            int lastx = <int> lastIndex;
+            lastx += 1;
+            foreach var i in 0...lastx{
+                byte remove = byteArray.remove(i);
+            }   
 
-                io:Error? fileWriteString2 = io:fileWriteString("file7.json",secondstring,io:APPEND);
+            io:println(byteArray);
 
-                // json|error j = secondstring.fromJsonString();
-                // io:println(j);
+            string|error firststring = string:fromBytes(data);
```

Comment:
This is a local variable, these are not preserved between massages so you'll have to use object field to store information that you need for the second message.